src/compiler/scala/tools/nsc/CompilationUnits.scala

```diff
@@ -68,10 +68,10 @@ trait CompilationUnits { global: Global =>
     /** Note: depends now contains toplevel classes.
      *  To get their sourcefiles, you need to dereference with .sourcefile
      */
-    private[this] val _depends = mutable.HashSet[Symbol]()
+    private[this] val _depends = if (settings.YtrackDependencies.value) mutable.HashSet[Symbol]() else null
     @deprecated("Not supported and no longer used by Zinc", "2.12.9")
-    def depends = _depends
-    def registerDependency(symbol: Symbol): Unit = {
+    def depends: mutable.HashSet[Symbol] = if (_depends == null) mutable.HashSet[Symbol]() else _depends
+    def registerDependency(symbol: Symbol): Unit = if (settings.YtrackDependencies.value) {
```

Comment:
not necessary to explicitly de-value a `BooleanSetting`

Reply:
I'm a bit wary of `implicit` `reflectSettingToBoolean` as it goes through Boolean box/unboxing. Boolean boxing doesn't actually allocate, but it it does incur a few extra branches.

Reply:
Kind of at the level of we can't have nice things. But I understand your sensitivity to the issue.